### DISCUSSION TITLE
about mobile development

### DISCUSSION THREAD

**Discussion by rraammiinn** [imba/imba#824] [Q&A]
can it be used for mobile apps development like reactnative and ionic ?

**Comment by haikyuu** [comment]
ionic is the go to solution. I highly recommend it

**Reply by rraammiinn** [reply]
can it be integrated with imba ?

**Reply by rraammiinn** [reply]
of course I don't care about ionic components ; I only care about making apk with native api access .

**Reply by haikyuu** [reply]
capacitor works with any web technology, and it's the one responsible for native access ...

Ionic components also work (a commit not published yet), and for routing it's possible to use ion-router as described in the docs (for web component use).

**Reply by rraammiinn** [reply]
what's important to me is a cli tool for building apk without android studio .
android studio is so heavy for my system .

**Reply by rraammiinn** [reply]
can you add capacitor template to imba cli as you did for tauri ?

**Reply by haikyuu** [reply]
Sure, it might take some time. I'd rather wait for the next imba version that has more solid Vite support

**Reply by rraammiinn** [reply]
thanks , that's great .
it's really hard to work simultaneously with capacitor cli and imba cli .
it's great if imba cli do both .

**Reply by haikyuu** [reply]
@rraammiinn 🤔 I'm also building an Imba app using capacitor and ionic . Could you elaborate on why it's hard to work with both capacitor and imba cli?

**Reply by rraammiinn** [reply]
it's mostly about the template that capacitor generates for you that is useless for imba and you should remove a lot of it .
I know it's ok for most of people but having a prepared template and using only one cli from one directory is very nice .

**Reply by rraammiinn** [reply]
it also took me a lot of time to figure out how to sign my app without android studio and I don't know why capacitor doesn't do it automatically when you run npx capacitor build android .

**Reply by rraammiinn** [reply]
it also can be great if android permissions be added automatically according to what apis you use in project .

**Reply by rraammiinn** [reply]
I'm also wondered if imba cli can extend capacitor cli so it can build project also for desktop using tauri .

**Reply by haikyuu** [reply]
I think these are concerns to be treated in capacitor

**Reply by haikyuu** [reply]
but a better template is definitely on the list

**Comment by taviks** [comment]
I'm using Imba with [Tauri](https://github.com/tauri-apps/tauri) for building a desktop app (which has overall been a good experience). Tauri Mobile is not yet production ready but did enter alpha on Dec 9, 2022 ([blog announcement here](https://tauri.app/blog/2022/12/09/tauri-mobile-alpha)).

**Reply by rraammiinn** [reply]
yes , that's amazing that imba has such a template for tauri but mobile development is game changing .
